haas/rest.py

```diff
@@ -209,9 +209,13 @@ def _rest_wrapper(f, schema):
 
     def wrapper(**kwargs):
         kwargs = _do_validation(schema, kwargs)
-        logger.debug('Got api call: %s(%s)' %
-                     (f.__name__, _format_arglist(**kwargs)))
+
         init_auth()
+        username = '(guest)' if not hasattr(local, 'auth') \
+            else flask.request.authorization.username
```

Comment:
Note that this line puts an additional constraint on auth methods----they must set flask.request.authorization.username .  We should document this.  We could also potentially have the null auth backend set username to (no user), and get rid of the special case